controllers/spiaccesstoken_controller.go

```diff
@@ -157,6 +157,24 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
+	if at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData {
+		func() {
+			hasLinkedBindings, err := hasLinkedBindings(ctx, &at, r.Client)
+			if err != nil {
+				lg.Error(err, "failed to validate the object")
+				return
+			}
+			if hasLinkedBindings {
+				return
+			}
+			//if token is in Awaiting, and no linked bindings present, it means that it have no bindings referring to it and can be cleaned up
+			if err := r.Delete(ctx, &at); err != nil {
+				lg.Error(err, "failed to cleanup the processed token")
+				return
+			}
+		}()
```

Comment:
This means that if we successfully delete the token in this anonymous function, we continue with the reconciliation, right? Not sure this is the correct logic. IMHO there is no need to "hide" the errors from the rest of the reconciliation function like this. The token is in a valid phase and we're reconciling that phase. If we fail, the reconciliation should fail and be retried (which happens automatically if you return an error).